File: PiNumber/Python/PiNumber.py
import random

def piNumber(N):

    T = 0

    for i in range(N):
        xCoord = random.uniform(0, 1);
        yCoord = random.uniform(0, 1);

        if(xCoord**2 + yCoord**2 <= 1):
            T += 1

    return 4 * T / N

print("%.50f" % piNumber(10000000))

File: PiNumber/C#/PiNumber.cs
using System;
using System.Numerics;

namespace PiNumber
{
	class Program 
	{
		public static double piNumber(double numberOfAllPoints)
		{
			double xCoord = 0.0, yCoord = 0.0, numberOfPointsInsideCircle = 0.0;
			
			Random rnd = new Random();
			
			for(int i = 0; i < numberOfAllPoints; i++)
			{

				xCoord = rnd.NextDouble();
				yCoord = rnd.NextDouble();

				if(xCoord * xCoord + yCoord * yCoord <= 1)
					numberOfPointsInsideCircle++;
			}

			return 4 * numberOfPointsInsideCircle / numberOfAllPoints;
		}

		public static void Main(string[] args)
		{
			Console.WriteLine(piNumber(10000000));	
		}
	}
}

